hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -518,11 +522,36 @@ private Map<String, String> convertToMetadata(Database database) {
     if (database.getDescription() != null) {
       meta.put("comment", database.getDescription());
     }
+    if (database.getOwnerName() != null) {
+      meta.put(TableProperties.HMS_DB_OWNER, database.getOwnerName());
+      if (database.getOwnerType() != null) {
+        meta.put(TableProperties.HMS_DB_OWNER_TYPE, database.getOwnerType().name());
+      }
+    }
 
     return meta;
   }
 
   Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(
```

Comment:
I think Precondition are meant to check whether user/caller invoked the method with the correct input/parameters.

Understandably, different sources will hold different opinions, but these are the sources I found when I tried to google around:

Quoting these two java tutorial sites https://www.geeksforgeeks.org/preconditions-guava-java/ & https://www.baeldung.com/guava-preconditions

"The Preconditions Class provides a list of static methods for checking that a method or a constructor is invoked with valid parameter values"

Quoting java doc for guava at https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html

"Non-preconditions
It is of course possible to use the methods of this class to check for invalid conditions which are not the caller's fault. Doing so is not recommended because it is misleading to future readers of the code and of stack traces. "

Reply:
Thanks for the explanation, @haizhou-zhao! We definitely use Preconditions in Impala for other purposes as we instead guarantee that the internal invariant is not hurt and we don't validate user input with them. But we might use it against the original purpose then :)
Thanks again for the explanation!